third_party/amd/lib/TritonAMDGPUTransforms/CanonicalizePointers.cpp

```diff
@@ -269,6 +285,39 @@ Value createTensorZero(IRRewriter &rw, Location loc, RankedTensorType type) {
 
 } // namespace
 
+void PointerCanonicalizer::collectFatPointerAttributes(Operation *op,
+                                                       Value val) {
+  auto addBlockArgumentAttr = [&](BlockArgument arg) {
+    // If the value is a block parameter, the operation can specify
+    // an attribute for the given parameter by using `tt.property_argi`
+    // where `argi` referrs to the arg number of the given parameter.
+    // So we need to iterato through the property, find the right one
+    // and push the property onto the pointers attributes.
+    for (NamedAttribute namedAttr : op->getAttrs()) {
+      llvm::SmallString<8> nameSuffix;
```

Comment:
This will allocate and deallocate in a loop. Maybe just move it outside and clear?

Reply:
Isn't that stack allocated?

Reply:
`SmallString` is a `SmallVector` under the hood right? `SmallVector` is a proper object which has a dynamic part that you need to alloc/dealloc.

Reply:
With `<8>`  SmallVector first has a data-member `T arr[8]` and it will use that until the size grows over 8 elements. So this string shouldn't cross that boundary with `_arg<index>`.

Reply:
Yeah. but you still need to init and deinit given this is a proper C++ object right? Maybe the original "allocate and deallocate" wording is confusing here.

Reply:
Yes, true. Not as bad as malloc but still a cost.

Reply:
Yeah anecdotal: I used to do such `SmallVector` in a loop and one day after profiling it turns out quite more expensive than I originally thought. of course depends on how hot the loop is so ymmv. just wanted to provide some context why i'm being a bit nitpicking here--been bitten by it before. :)

Reply:
Interesting discussion! Yeah I agree that moving the construction out is a good idea :) ! Thanks guys!